### PR TITLE
[3006.x][DOC] add note for reactor state id uniquness. 

### LINKED ISSUE #63589: [DOCS] reactors should have unquie state ids to avoid nameing conflicts.
**Description**
currently the documentation for reactors does not call out that reactors should have unique state ids. this normally might not be a problem in most cases. however it is possible to have a name collision. which will cause reactors to silently fail or not run. 

the possibility of name collision is called out in the code. at https://github.com/saltstack/salt/blob/4ec04292e5da528c5dc8b93ed2455a200589e08f/salt/utils/process.py#L448-L450

**Suggested Fix**
document the possibility of name collision causing dropped reactors.

**Location or format of documentation**
https://docs.saltproject.io/en/latest/topics/reactor/index.html

**Additional context**
This can cause an almost undetectable issue where a reactor will just not run and no error messages being displayed.

### PR DESCRIPTION
**Pull request by whytewolf**
### What does this PR do? 

doc update to fix missing note about unique state ids for runner and wheel reactors. 

### What issues does this PR fix or reference?
Fixes: #63589 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### CODE REVIEW

**Review by s0undt3ch — DISMISSED** [review]

File: doc/topics/reactor/index.rst [hunk 1]
```diff
@@ -212,6 +212,10 @@ in :ref:`local reactions <reactor-local>`, but as noted above this is not very
 user-friendly. Therefore, the new config schema is recommended if the master
 is running a supported release.
 
+.. note::
+  state ids of reactors for runners and wheel should all be unique. they can
```
Comment (twangboy) [inline doc/topics/reactor/index.rst:216]: State ids... wheels... They...

File: doc/topics/reactor/index.rst [hunk 2]
```diff
@@ -212,6 +212,10 @@ in :ref:`local reactions <reactor-local>`, but as noted above this is not very
...
+  overwrite each other when added to the async queue. causing lost reactions
```
Comment (twangboy) [inline doc/topics/reactor/index.rst:217]: Remove the period after queue... add a period at the end.

File: doc/topics/reactor/index.rst [hunk 3]
```diff
@@ -248,6 +252,10 @@ Like :ref:`runner reactions <reactor-runner>`, the old config schema called for
 wheel reactions to have arguments passed directly under the name of the
 :ref:`wheel function <all-salt.wheel>` (or in ``arg`` or ``kwarg`` parameters).
 
+.. note::
+  state ids of reactors for runners and wheel should all be unique. they can
+  overwrite each other when added to the async queue. causing lost reactions
```
Comment (twangboy) [inline doc/topics/reactor/index.rst:257]: Same stuff in this paragraph.

**Review by twangboy — CHANGES_REQUESTED** [review]

File: doc/topics/reactor/index.rst (hunk 1 above)
Comment (whytewolf) [reply doc/topics/reactor/index.rst]: done

File: doc/topics/reactor/index.rst (hunk 3 above)
Comment (whytewolf) [reply doc/topics/reactor/index.rst]: done

File: doc/topics/reactor/index.rst (hunk 2 above)
Comment (whytewolf) [reply doc/topics/reactor/index.rst]: done

File: doc/topics/reactor/index.rst [hunk 4]
```diff
@@ -212,6 +212,10 @@ in :ref:`local reactions <reactor-local>`, but as noted above this is not very
 user-friendly. Therefore, the new config schema is recommended if the master
 is running a supported release.
 
+.. note::
+  State ids of reactors for runners and wheels should all be unique.Tthey can
```
Comment (twangboy) [inline doc/topics/reactor/index.rst:216]: Tthey

File: doc/topics/reactor/index.rst [hunk 5]
```diff
@@ -212,6 +212,10 @@ in :ref:`local reactions <reactor-local>`, but as noted above this is not very
...
+  overwrite each other when added to the async queue causing lost reactions
```
Comment (twangboy) [inline doc/topics/reactor/index.rst:217]: End with a period

File: doc/topics/reactor/index.rst [hunk 6]
```diff
@@ -248,6 +252,10 @@ Like :ref:`runner reactions <reactor-runner>`, the old config schema called for
 wheel reactions to have arguments passed directly under the name of the
 :ref:`wheel function <all-salt.wheel>` (or in ``arg`` or ``kwarg`` parameters).
 
+.. note::
+  State ids of reactors for runners and wheels should all be unique. They can
+  overwrite each other when added to the async queue causing lost reactions
```
Comment (twangboy) [inline doc/topics/reactor/index.rst:257]: End with a period

**Review by twangboy — CHANGES_REQUESTED** [review]

**Review by twangboy — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**